### PR TITLE
Update SDK to pick up incremental Razor source generator

### PR DESCRIPTION
**Pull request by captainsafia**

### CODE REVIEW

**Comment by BrennanConroy** [comment]
We've been trying to update the SDK :/
https://github.com/dotnet/aspnetcore/pull/33719

**Comment by captainsafia** [comment]
@BrennanConroy Ack. I'm doing this to shake out any issues with the incremental source generator that I just merged. I mostly want to verify that the builds for projects with Razor/CSHTML are fine.

File: src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj [hunk 1]
```diff
@@ -8,7 +8,6 @@
       Clien caching isn't part of our performance measurement, so we'll skip it.
     -->
     <BlazorCacheBootResources>false</BlazorCacheBootResources>
-    <UseRazorSourceGenerator>false</UseRazorSourceGenerator>
```
Comment (captainsafia) [inline src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj:11]: We don't need this anymore. I fixed the root cause in https://github.com/dotnet/aspnetcore/pull/33456 anyways.

File: src/Components/test/testassets/BasicTestApp/BasicTestApp.csproj [hunk 2]
```diff
@@ -11,7 +11,6 @@
 
     <!-- Project supports more than one language -->
     <BlazorWebAssemblyLoadAllGlobalizationData>true</BlazorWebAssemblyLoadAllGlobalizationData>
-    <UseRazorSourceGenerator>false</UseRazorSourceGenerator>
```
Comment (captainsafia) [inline src/Components/test/testassets/BasicTestApp/BasicTestApp.csproj:14]: With local and CI-based runs of `./build.sh` in `src/Components` I'm not seeing https://github.com/dotnet/aspnetcore/issues/31023 anymore so I think we're good here.

If that turns out not to be the case, we can revisit.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**